src/Dev/CliDebugView.php

```diff
@@ -129,4 +131,76 @@ public function renderVariable($val, $caller)
 
         return $output;
     }
+
+    /**
+     * Similar to renderVariable() but respects debug() method on object if available
+     *
+     * @param mixed $val
+     * @param array $caller
+     * @param bool $showHeader
+     * @return string
+     */
+    public function debugVariable($val, $caller, $showHeader = true)
+    {
+        $text = $this->debugVariableText($val);
+        if ($showHeader) {
+            $callerFormatted = $this->formatCaller($caller);
+            return "Debug ($callerFormatted)\n{$text}\n\n";
+        } else {
+            return $text;
+        }
+    }
+
+    /**
+     * Get debug text for this object
+     *
+     * @param mixed $val
+     * @return string
+     */
+    public function debugVariableText($val)
+    {
+        // Check debug
+        if (ClassInfo::hasMethod($val, 'debug')) {
+            return $val->debug();
```

Comment:
Unfortunately debug() doesn't allow context to be provided; We simply (and have to) assume that user-implemented debug() methods will output in the correct format.

User code can use Debug::create_debug_view() internally to achieve this, but we can't easily and probably shouldn't enforce it.